### PR TITLE
[RFC] nvim_list_uis: include channel id

### PR DESCRIPTION
**Pull request by bfredl**
Some minor fixes for `nvim_list_uis()`

- Currently there is no way to tell what channel is associated with what UI in the list. (lack of channel also implicitly identifies TUI, though we could add "term" key)
- Docs for `nvim_list_uis()` are a bit lacking
- `nvim_list_uis()[0].rgb` didn't reflect changes to `termguicolors` when using TUI.

Nominating for 0.3, ping @justinmk.

### CODE REVIEW

File: src/nvim/ui.c [hunk 1]
```diff
@@ -535,14 +535,18 @@ Array ui_array(void)
...
   for (size_t i = 0; i < ui_count; i++) {
-    Dictionary dic = ARRAY_DICT_INIT;
-    PUT(dic, "width", INTEGER_OBJ(uis[i]->width));
-    PUT(dic, "height", INTEGER_OBJ(uis[i]->height));
-    PUT(dic, "rgb", BOOLEAN_OBJ(uis[i]->rgb));
+    UI *ui = uis[i];
+    Dictionary info = ARRAY_DICT_INIT;
+    PUT(info, "width", INTEGER_OBJ(ui->width));
+    PUT(info, "height", INTEGER_OBJ(ui->height));
+    PUT(info, "rgb", BOOLEAN_OBJ(ui->rgb));
```
Comment (justinmk) [inline src/nvim/ui.c]: should `inspect()` also set these other properties?

Comment (bfredl) [reply src/nvim/ui.c]: why? they are shared amongst all UI:s, so what would be the point of duplicating them?

Comment (justinmk) [reply src/nvim/ui.c]: other way round then: why do we need inspect()? Is it not possible to get the channel of a UI from here?

Comment (bfredl) [reply src/nvim/ui.c]: the channel is specific to one UI kind, the rest of the fields are generic. Other way around: why should this method deviate from how all other UI functionality delegates from general to specific implementation?

Comment (bfredl) [reply src/nvim/ui.c]: Also I see room for futher expansion, such as terminfo info from the TUI, but that is post-0.3 to me.

**Comment by justinmk** [comment]
QB looks worse than usual, might want to poke it again.

**Comment by bfredl** [comment]
@justinmk it seems TUI/bridge teardown doesn't respect event ordering, but uses a side-channel. Maybe we don't want to refactor that now; we could set `brigde->rgb` using the bridge private mutex also, though I think that would technically (thus, actually) be a race condition... Maybe let's cheat and let the bridge handle `termguicolors` also for now (0.3).

**Pull request merged**